tests/example.py

```diff
@@ -0,0 +1,143 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script implemets Puppeteer's `examples/cross-browser.js` scenario using WebDriver BiDi.
+# https://github.com/puppeteer/puppeteer/blob/4c3caaa3f99f0c31333a749ec50f56180507a374/examples/cross-browser.js
+
+import asyncio
+import json
+import os
+import pytest
+import websockets
+import time
+
+async def get_websocket():
+    port = os.getenv('PORT', 8080)
+    url = f'ws://localhost:{port}'
+    return await websockets.connect(url)
+
+async def send_JSON_command(websocket, command):
+    await websocket.send(json.dumps(command))
+
+async def read_JSON_message(websocket):
+    return json.loads(await websocket.recv())
+
+async def main():
+    # const browser = await puppeteer.launch();
+    # https://github.com/puppeteer/puppeteer/blob/4c3caaa3f99f0c31333a749ec50f56180507a374/examples/cross-browser.js#L29
+    websocket = await get_websocket()
+
+    # Receive initial `browsingContext.contextCreated` event.
+    resp = await read_JSON_message(websocket)
+    assert resp['method'] == 'browsingContext.contextCreated'
+
+    # Not implemented:
+    # await browser.version());
+    # https://github.com/puppeteer/puppeteer/blob/4c3caaa3f99f0c31333a749ec50f56180507a374/examples/cross-browser.js#L32
+
+    # Puppeteer:
+    # const page = await browser.newPage();
+
+    # Part 1. Execute BiDi command.
+    # ... browser.newPage();
+    # https://github.com/puppeteer/puppeteer/blob/4c3caaa3f99f0c31333a749ec50f56180507a374/examples/cross-browser.js#L31
+    command = {
+        "id": 1000,
+        "method": "PROTO.browsingContext.create",
+        "params": {}}
+    await send_JSON_command(websocket, command)
+
+    # Receive `browsingContext.contextCreated` event.
+    resp = await read_JSON_message(websocket)
+    assert resp['method'] == 'browsingContext.contextCreated'
+
+    # Part 2. Wait for result.
+    # ... await ...
+    # https://github.com/puppeteer/puppeteer/blob/4c3caaa3f99f0c31333a749ec50f56180507a374/examples/cross-browser.js#L31
+    resp = await read_JSON_message(websocket)
+    assert resp["id"] == 1000
+
+    # Part 3. Get the command result.
+    # const page = ...
+    # https://github.com/puppeteer/puppeteer/blob/4c3caaa3f99f0c31333a749ec50f56180507a374/examples/cross-browser.js#L31
+    contextID = resp['result']['context']
+
+    # Puppeteer:
+    # const page = await browser.newPage();
+
+    # Part 1. Execute BiDi command.
+    # ... page.goto('https://news.ycombinator.com/');
+    # https://github.com/puppeteer/puppeteer/blob/4c3caaa3f99f0c31333a749ec50f56180507a374/examples/cross-browser.js#L34
+    command = {
+        "id": 1001,
+        "method": "browsingContext.navigate",
+        "params": {
+            "url": "https://news.ycombinator.com/",
```

Comment:
Note that hitting https://news.ycombinator.com/ in the test will likely make the test flaky, as it probably doesn't have 100% uptime.

Reply:
it's not a test, but an example script. Probably should be moved to another folder.

Reply:
which I just did

Reply:
Oh :)